firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -273,12 +283,20 @@ private void executeFidRegistration(PersistedFidEntry persistedFidEntry, AwaitLi
     }
   }
 
-  private void updatePersistedFidWithPendingStatus(String fid) {
+  /**
+   * Refreshes FID auth token with FIS servers in a background thread and updates the listener on
+   * completion.
+   */
+  private void refreshAuthToken(AwaitListener listener) {
+    Task<InstallationTokenResult> task = Tasks.call(executor, () -> refreshAuthTokenIfNecessary(1));
+    if (listener != null) {
+      task.addOnCompleteListener((unused) -> listener.onSuccess());
+    }
+  }
+
+  private void updatePersistedFidWithPendingStatus(PersistedFidEntry persistedFidEntry) {
```

Comment:
A general comment : I feel like you are doing functionalization too much in this class.

Functionalization is useful when you want to hide a chunk of detailed logic, or to avoid duplicated code. But for example for updatePersistedFidWithPendingStatus, this function is only used in one place(doesn't avoid duplicated code), and it's not hiding any detailed logic (use one line to replace another line)

A downside of over functionalization is, with too many small method, it's really hard to find the relationship between them.

Above all, for these cases, I prefer making the code inline than making as a function.

Reply:
Thanks for the detailed explanation. Updated the code. PTAL.